core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1223,4 +1223,98 @@ public void testMoveBetweenStructsFails() {
                 .moveBefore("s2.x", "s1.a")
                 .apply());
   }
+
+  @Test
+  public void testAddRowIdentifier() {
+    Schema newSchema = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
+        .allowIncompatibleChanges()
+        .addRowIdentifier("id")
+        .addRequiredColumn("id2", Types.StringType.get())
+        .addRowIdentifier("id2")
+        .apply();
+    Assert.assertEquals(Sets.newHashSet(1, SCHEMA_LAST_COLUMN_ID + 1), newSchema.rowIdentifiers());
+
+    AssertHelpers.assertThrows("Should fail if column with name not exist",
+        IllegalArgumentException.class,
+        "Cannot find column of the given name in existing or newly added columns",
+        () -> new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID).addRowIdentifier("unknown").apply());
+
+    AssertHelpers.assertThrows("Should fail if column with name not exist",
+        IllegalArgumentException.class,
+        "it is not a required field",
+        () -> new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID).addRowIdentifier("data").apply());
+
+    AssertHelpers.assertThrows("Should fail if column with name not exist",
+        IllegalArgumentException.class,
+        "it is not a primitive field",
+        () -> new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID).addRowIdentifier("locations").apply());
+  }
+
+  @Test
+  public void testRenameRowIdentifier() {
+    Schema schemaWithRowIdentifier = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
+        .allowIncompatibleChanges()
+        .addRowIdentifier("id")
+        .apply();
+
+    Schema newSchema = new SchemaUpdate(schemaWithRowIdentifier, SCHEMA_LAST_COLUMN_ID)
+        .renameColumn("id", "id2")
+        .apply();
+
+    Assert.assertEquals(Sets.newHashSet(1), newSchema.rowIdentifiers());
+  }
+
```

Comment:
What's the behavior if we try to add a unit tests based on this [comment](https://github.com/apache/iceberg/pull/2465/files#r612199953) ?

Reply:
I think what you describe is `Should fail if deleting column before deleting it as row identifier` case in `testDeleteRowIdentifier` test